### ISSUE TITLE
Detect constant conditions in `match`

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#3207]
This `match` condition does not report any violations right now:

```python
match 1:
    case SomeClass(field): ...
```

However, this does report a violation:

```python
if 1: ...
```

```
  1:4      WPS314 Found conditional that always evaluates the same
  if 1: ...
     ^
```

What needs to be done?
- Change docs here: https://github.com/wemake-services/wemake-python-styleguide/blob/e878f3a0cd17c612ad1bc4b442bdcd1c005d20c2/wemake_python_styleguide/violations/consistency.py#L664
- Add new `visit_Match` to https://github.com/wemake-services/wemake-python-styleguide/blob/e878f3a0cd17c612ad1bc4b442bdcd1c005d20c2/wemake_python_styleguide/visitors/ast/compares.py#L169

with something like:

```python
    def visit_Match(self, node: ast.Match) -> None:
        self._check_constant_condition(node.subject)
        self.generic_visit(node)
```
- Add tests to https://github.com/wemake-services/wemake-python-styleguide/blob/master/tests/test_visitors/test_ast/test_compares/test_conditionals.py
- Update `CHANGELOG.md`
